lib/mongo/server/connection_base.rb

```diff
@@ -213,12 +212,21 @@ def serialize(message, client, buffer = BSON::ByteBuffer.new)
           max_bson_size += MAX_BSON_COMMAND_OVERHEAD
         end
 
-        final_message.serialize(buffer, max_bson_size)
+        temp_buffer = BSON::ByteBuffer.new
+        temp_buffer.put_string(buffer.to_s)
```

Comment:
I believe this should be `put_bytes` and could you add a comment that this copies the existing passed in buffer contents, if any.